### DISCUSSION TITLE
Validações

### DISCUSSION THREAD

**Discussion by depaula36** [balta-io/forum#620] [Tira Dúvidas] [answered]
Professor,
Quando faço a validação (mais de um validador) com Data Annotation no model, só é exibida UMA das mensagens de erro na View (na View uso <span asp-validation-for="<nome da propriedade do model"></span><br/>)
Obs; Tentei usar a estrutura acima dentro de um foreach, mas só repete a mesma mensagem.
Tem algum segredo?
Desde já agradeço.

**Comment by depaula36** [comment]
Na view uso:
<span asp-validation-for="Nome"></span><br/>

**Comment by depaula36** [comment]
(<span asp-validation-for="Nome"></span><br/>)

**Comment by depaula36** [comment]
span asp-validation-for="Nome"></span><br/

**Comment by depaula36** [comment]
Onde nome é nome da propriedade com os validadores (tirei o < e > para poder imprimir o trecho span pq estava sumindo ao colar aqui...rsrs

**Comment by depaula36** [comment]
Consgui, Professor:
div asp-validation-summary="All">
                <span asp-validation-for="Nome"></span>
</div

**Comment by depaula36** [comment]
Consegui, professor:
<img width="257" alt="image" src="https://github.com/user-attachments/assets/8b7a8c0f-3fa2-4b90-ad4b-48448875d6c2">

**Comment by depaula36** [comment]
Ainda não é isso (o sumary traz de todos os campos e só quero de uma propriedade do model) (e esse span de dentro também nem devia existir no na tela colada acima)

**Comment by depaula36** [comment, marked as answer]
Professor, vai dar certo por aqui usando (vou conseguir capturar usando as propriedades do ModelState):
<img width="329" alt="image" src="https://github.com/user-attachments/assets/040cc51c-a7c3-4298-949b-814069485b24">

Desculpe a confusão das mensagens..rssr..Grande abraço.

**Reply by andrebaltieri** [reply]
Tranquilo 🚀
